app/web/src/newhotness/Explore.vue

```diff
@@ -660,17 +660,9 @@ const componentList = computed(() => {
   if (!pinnedComponent.value || !pinnedComponentConnections.value)
     return componentListUnchecked.value;
 
-  // If we are dealing with pinning, we are only concerned with relevant components.
-  const relevantComponentsWhenPinning = [pinnedComponent.value];
-  for (const component of componentListUnchecked.value) {
-    if (
-      pinnedComponentConnectionSets.value.incoming.has(component.id) ||
-      pinnedComponentConnectionSets.value.outgoing.has(component.id)
-    ) {
-      relevantComponentsWhenPinning.push(component);
-    }
-  }
-  return relevantComponentsWhenPinning;
+  // When pinning, include all components so they can be properly grouped
+  // into connected and unconnected sections
+  return componentListUnchecked.value;
```

Comment:
We may be able to simplify all of this in the future, but I agree with keeping the list and unchecked list separate for now. The second that we have a product need for a list of components that is smaller than the full list, we'll want this separation